### PR TITLE
Feature/Ability to interact with DNS message question/answer context

### PR DESCRIPTION
**Pull request by bestwebua**
- [x] Added `DnsMock::Server#messages`, `DnsMock::Server#clear_messages!`, tests
- [x] Updated `DnsMock::Server#run`, tests
- [x] Updated integration tests
- [x] Updated project documentation

### CODE REVIEW

**Comment by bestwebua** [comment]
@mroach Please, take a look. Does it fit for your?

File: lib/dns_mock/response/message.rb [hunk 1]
```diff
@@ -24,10 +28,15 @@ def as_binary_string
...
       attr_reader :dns_answer, :dns_message
+      attr_writer :lookup
 
       def compose_answer
         dns_message.each_question do |hostname, record_type|
-          dns_message.answer.push(*dns_answer.build(hostname, record_type))
+          self.lookup = DnsMock::Response::Message::Lookup.new(
+            question: [hostname, record_type],
+            answer: dns_answer.build(hostname, record_type)
+          )
```
Comment (mroach) [inline lib/dns_mock/response/message.rb:38]: This causes `lookup` to be overwritten with each loop iteration. I'm not familiar enough with the inner-workings of DNS. Is there only ever one query per "message"?

Comment (bestwebua) [reply lib/dns_mock/response/message.rb:38]: As I see this array includes only one element. Could you run something to get more then one?

**Comment by bestwebua** [comment]
@mroach What about simplifying? In `server.messages` we can return just `dns_message`  instances (`Resolv::DNS::Message`), that already included examples of requests and responses.

**Pull request merged**